test-data/genops-basic.test

```diff
@@ -973,3 +973,112 @@ L0:
     r1 = f() :: object
     r2 = cast(float, r1)
     return r2
+
+[case testNestedFunctions]
+from typing import Callable
+
+def outer() -> Callable[[], str]:
+    def inner() -> str:
+        return 'inner: nested function'
+    return inner
+
+def first() -> Callable[[], Callable[[], str]]:
+    def second() -> Callable[[], str]:
+        def third() -> str:
+            return 'third: nested function'
+        return third
+    return second
+
+def uno(num: float) -> Callable[[str], str]:
+    def dos(s: str) -> str:
+        return s + '!'
+    return dos
+
+def eins(num: float) -> str:
+    def zwei(s: str) -> str:
+        return s + '?'
+    return zwei('eins')
```

Comment:
Add a test that calls an inner function twice, so we can verify that the object only gets constructed once (after that fix is made)

Reply:
Done.